### PR TITLE
DashR Drug Discovery

### PR DESCRIPTION
**Pull request by mleech1**
Issue for app: #[issue number here]

# App pull request
- [x] This is a new app
- [x] I am improving an existing app (redesigns/code "makeovers")

## About

- Playground deployment URL (new version):
- Current gallery app URL: (delete this line if inapplicable)
- Python app repository link: (delete this line if you are working on a Python app)

## Workflow

- [x] I have created a branch in the appropriate monorepo, and the
  elements necessary for successful deployment are in place.
- [ ] If the app is a redesigned and/or restyled version of an
  existing gallery app, I've summarized the changes requested in the
  appropriate Streambed issue and confirm that they have been applied.
- [ ] If the app is on the Dash Gallery portal, I have added a link to
  the GitHub repository for the source code in the portal description.
- [ ] If the app is a reimplementation of a Python gallery app for the
  DashR gallery, the app in this PR mimics, as closely as possible,
  the style and functionality of the existing app.=
- [ ] I have removed *all Google Analytics code* from the app's
  `assets/` folder.

## The pre-review review

I have addressed all of the following questions:

- [ ] Does everything in my code serve some purpose? (I have removed
  any dead and/or irrelevant code.)
- [ ] Does everything in my code have a clear purpose? (My code is
  readable and, where it isn't, it has been commented appropriately.)]
- [ ] Am I reinventing the wheel? (I have used appropriate packages to
  lessen the volume of code that needs to be maintained.)

### CODE REVIEW

File: apps/dashr-drug-discovery/app.R [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    return(name)
+  }
+)
+
+#append to table
+app$callback(
+  output = list(id = 'table-element', property = 'children'),
+  params = list(input(id = 'chem_dropdown', property ='value')),
+  function(chem_dropdown_value){
+    table1 = make_dash_table(chem_dropdown_value)
```
Comment (z267xu) [inline apps/dashr-drug-discovery/app.R]: `<-`

File: apps/dashr-drug-discovery/app.R [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+  }
+)
+
+#update name
+app$callback(
+  output = list(id = 'chem_name', property = 'children'),
+  params = list(input(id = 'clickable-graph', property = 'hoverData')),
+  function(hover){
+    info <- hover$points[[1]]$text
+    name = df$NAME[df$NAME==info]
```
Comment (z267xu) [inline apps/dashr-drug-discovery/app.R]: `<-`

File: apps/dashr-drug-discovery/app.R [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+  }
+)
+
+#callback for name
+app$callback(
+  output = list(id = 'chem_name', property = 'children'),
+  params = list(input(id = 'clickable-graph', property = 'hoverData')),
+  function(hover){
+    info <- hover$points[[1]]$text
+    page_url = df$PAGE[df$NAME==info]
```
Comment (z267xu) [inline apps/dashr-drug-discovery/app.R]: `<-`

File: apps/dashr-drug-discovery/app.R [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+  }
+)
+
+#call back from img
+app$callback(
+  output = list(id = 'chem_img', property = 'src'),
+  params = list(input(id = 'clickable-graph', property = 'hoverData')),
+  function(hover){
+    info <- hover$points[[1]]$text
+    img = df$IMG_URL[df$NAME == info]
```
Comment (z267xu) [inline apps/dashr-drug-discovery/app.R]: `<-`

File: apps/dashr-drug-discovery/app.R [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+  }
+)
+
+#callback for description
+app$callback(
+  output = list(id = 'chem_desc', property = 'children'),
+  params = list(input(id = 'clickable-graph', property = 'hoverData')),
+  function(hover){
+    info <- hover$points[[1]]$text
+    description = df$DESC[df$NAME == info]
```
Comment (z267xu) [inline apps/dashr-drug-discovery/app.R]: `<-`

File: apps/dashr-drug-discovery/app.R [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+#get img url from csv, initalized with starting drug
+DRUG_DESCRIPTION = df$DESC[df$NAME == STARTING_DRUG]
+#get desc from csv, initalized with starting drug
+DRUG_IMG = df$IMG_URL[df$NAME == STARTING_DRUG]
+
+app$layout(
+  htmlDiv(list(
+    htmlDiv(list(
+      #Dash icon
+      htmlImg(src="https://s3-us-west-1.amazonaws.com/plotly-tutorials/logo/new-branding/dash-logo-by-plotly-stripe.png",
```
Comment (rpkyle) [inline apps/dashr-drug-discovery/app.R]: 🚚 `assets` folder 🙂

File: apps/dashr-drug-discovery/app.R [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+          children=make_dash_table(STARTING_DRUG),
+          className = "table__container"
+        )
+      ), className = "container bg-white p-0")
+    ),className = "app__container")
+  ))
+)
+
+
+#####CALL BACKS#####
```
Comment (rpkyle) [inline apps/dashr-drug-discovery/app.R]: Maybe rename `CALL BACKS` to `CALLBACKS`? I realize it's a comment, being nit picky here.

**Comment by rpkyle** [comment]
As we discussed offline, you'll need to change

```
df <- read.csv('/dashr-drug-discovery/data/small_molecule_drugbank.csv', header = TRUE, sep = ",")
```

to

```
df <- read.csv('./data/small_molecule_drugbank.csv', header = TRUE, sep = ",")
```

since `setwd` already places us in the `dashr-drug-discovery` directory 🙂

**Comment by rpkyle** [comment]
You'll also need three more deployment-related files in your `dashr-drug-discovery` directory:

`Aptfile`
- `.buildpacks`
- `init.R`

**Comment by rpkyle** [comment]
@mleech1 OK, unfortunately, still not ready to merge; need to address a few warnings:

**3D scatter, 2D scatter, 2D histogram on initial load of callback:**
```
warning: `line.width` does not currently support multiple values. from NULL
```

**On initial load of 2D histogram:**
```
warning: `line.width` does not currently support multiple values. from NULL
warning: 'histogram2d' objects don't have these attributes: 'text'
Valid attributes include:
'type', 'visible', 'showlegend', 'legendgroup', 'opacity', 'name', 'uid', 'ids', 'customdata', 'selectedpoints', 'hoverinfo', 'hoverlabel', 'stream', 'transforms', 'uirevision', 'x', 'y', 'z', 'marker', 'histnorm', 'histfunc', 'nbinsx', 'xbins', 'nbinsy', 'ybins', 'autobinx', 'autobiny', 'xgap', 'ygap', 'zsmooth', 'zhoverformat', 'hovertemplate', 'zauto', 'zmin', 'zmax', 'zmid', 'colorscale', 'autocolorscale', 'reversescale', 'showscale', 'colorbar', 'xcalendar', 'ycalendar', 'xaxis', 'yaxis', 'idssrc', 'customdatasrc', 'hoverinfosrc', 'xsrc', 'ysrc', 'zsrc', 'hovertemplatesrc', 'key', 'set', 'frame', 'transforms', '_isNestedKey', '_isSimpleKey', '_isGraticule', '_bbox'
 from NULL
warning: 'scatter' objects don't have these attributes: 'colorscale', 'showscale'
Valid attributes include:
'type', 'visible', 'showlegend', 'legendgroup', 'opacity', 'name', 'uid', 'ids', 'customdata', 'selectedpoints', 'hoverinfo', 'hoverlabel', 'stream', 'transforms', 'uirevision', 'x', 'x0', 'dx', 'y', 'y0', 'dy', 'stackgroup', 'orientation', 'groupnorm', 'stackgaps', 'text', 'hovertext', 'mode', 'hoveron', 'hovertemplate', 'line', 'connectgaps', 'cliponaxis', 'fill', 'fillcolor', 'marker', 'selected', 'unselected', 'textposition', 'textfont', 'r', 't', 'error_x', 'error_y', 'xcalendar', 'ycalendar', 'xaxis', 'yaxis', 'idssrc', 'customdatasrc', 'hoverinfosrc', 'xsrc', 'ysrc', 'textsrc', 'hovertextsrc', 'hovertemplatesrc', 'textpositionsrc', 'rsrc', 'tsrc', 'key', 'set', 'frame', 'transforms', '_isNestedKey', '_isSimpleKey', '_isGraticule', '_bbox'
 from NULL
```

**Related to hoverdata outside of window (as we discussed):**

```
error: subscript out of bounds
request: 127.0.0.1 - ID_127.0.0.1 [28/Jun/2019:11:51:49 -0400] "POST /_dash-update-component HTTP/1.1" 500 0 "http://127.0.0.1:8050/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_4) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/11.1 Safari/605.1.15"
```

For the last one, I'll help sort out a workaround. I suspect that we could set a default that is non-null, or edit the callback to address this gracefully.

**Comment by rpkyle** [comment]
Awesome! All fixed now! 💃 💃 💃

**Pull request merged**